### PR TITLE
Territorial peaceful

### LINKED ISSUE #3600: Territorial Time: End game ahead of the 4 hour mark when in Peaceful Mode
**Is your feature request related to a problem? Please describe.**
Territorial Time (TT) ends after 4 hours when none of the parties has been able to conquer more than 50% of the land for at least 20 minutes. Since we allow Peaceful Mode for TT, it may happen that players have conquered all available land, but since they are not allowed to attack, they have to wait for the 4hours to end.

**Describe the solution you'd like**
When one party has more land than other parties conquer so that they weill never overturn the leading one, end the game ahead of time.

Eg. party A has 42% party B has 38 and C party 19%. The game should be ended because neither B nor C can conquer enough land to overtake party A.

**Implementation steps**
- [ ] use the existing check for land, to determine whether a party has more land than is left for others to conquer and end the game

### PR DESCRIPTION
**Pull request by hessenfarmer**
fix #3600

### CODE REVIEW

**Review by stonerl — DISMISSED** [review]
LGTM

**Comment by stonerl** [comment]
We might consider removing the limitation for Territorial Lord as well.

**Comment by gunchleoc** [comment]
The changed code is common code for both territorial lord and territorial time.

**Comment by stonerl** [comment]
Yes but in Lord you cannot select peaceful ATM.

**Comment by gunchleoc** [comment]
+1 for allowing that before we merge this branch

**Comment by hessenfarmer** [comment]
Ok I will do so. However this type of game is totally different from a normal widelands game as it is all about expansion nothing else. so any ressources spent on other buildings then needed for construction is more the less a waste. It is just a race game in peaceful.

**Comment by gunchleoc** [comment]
Some people enjoy race games, so I'm find with that.

File: data/scripting/win_conditions/territorial_functions.lua [hunk 1]
```diff
@@ -86,7 +86,7 @@ function calculate_territory_points(fields, players)
 
    -- Check if we have a winner. The table was sorted, so we can simply grab the first entry.
    local winning_points = -1
-   -- Peaceful mode in territorial time
+   -- Peaceful mode needs more land then any other can gain (lead points > remaining fields)
```
Comment (gunchleoc) [inline data/scripting/win_conditions/territorial_functions.lua]: then -> than

File: data/scripting/win_conditions/territorial_lord.lua [hunk 2]
```diff
@@ -20,13 +20,15 @@ local wc_version = 2
 local wc_desc = _ (
    "Each player or team tries to obtain more than half of the map’s " ..
    "area. The winner will be the player or the team that is able to keep " ..
-   "that area for at least 20 minutes."
+   "that area for at least 20 minutes. " ..
+   "If the peaceful mode is selected the game ends if one player has more " ..
+   "land then any other player could gain."
```
Comment (gunchleoc) [inline data/scripting/win_conditions/territorial_lord.lua]: then -> than

File: data/scripting/win_conditions/territorial_time.lua [hunk 3]
```diff
@@ -24,7 +24,9 @@ local wc_desc = _ (
    "Each player or team tries to obtain more than half of the map’s " ..
    "area. The winner will be the player or the team that is able to keep " ..
    "that area for at least 20 minutes, or the one with the most territory " ..
-   "after 4 hours, whichever comes first."
+   "after 4 hours, whichever comes first." ..
+   "If the peaceful mode is selected, the game ends if one player has more " ..
+   "land then any other player could gain."
```
Comment (gunchleoc) [inline data/scripting/win_conditions/territorial_time.lua]: then -> than

**Review by gunchleoc — APPROVED** [review]
Tested and good to go :)

**Pull request merged**